backend/functions/views.py

```diff
@@ -1,30 +1,178 @@
+                if isinstance(translations_output, Exception):
+                    return translations_output
+
+                # Collect the translated sentences
+                translated_sentences = [
+                    translation["target"] for translation in translations_output
+                ]
+
+            elif api_type == "google":
+
+                # Get the translation using the Indictrans NMT API
+                translations_output = get_batch_translations_using_google_translate(
+                    sentence_list=batch_of_input_sentences,
+                    target_language=output_language,
+                )
+
+                # Check if translation output didn't return an error
+                if isinstance(translations_output, Exception):
+                    return translations_output
+
+                # Collect the translated sentences
+                translated_sentences = [
+                    translation["translatedText"] for translation in translations_output
+                ]
+
+            else:
+                return {"error": "Invalid API type. Allowed - [indic-trans, google]"}
+
+            # Check if the translated sentences are equal to the input sentences
+            if len(translated_sentences) != len(batch_of_input_sentences):
+                return {
+                    "error": "The number of translated sentences does not match the number of input sentences."
+                }
+
+            # Get the output dataset instance
+            output_dataset_instance = dataset_models.DatasetInstance.objects.get(
+                instance_id=output_dataset_instance_id
+            )
+
+            # Iterate through the dataframe
+            for index, row in input_sentences_df[i : i + batch_size].iterrows():
+
+                # Get the values for the TranslationPair model
+                sentence_text_id = row["sentence_text_id"]
+                input_sentence = row["corrected_text"]
+                input_language = row["input_language"]
+                translated_sentence = translated_sentences[index - i]
+                metadata = row["metadata"]
+                context = row["context"]
+
+                # Get the sentencetext model object by ID
+                sentence_text_object = dataset_models.SentenceText.objects.get(
+                    id=sentence_text_id
+                )
+
+                # Create and save a TranslationPair object
+                translation_pair_obj = dataset_models.TranslationPair(
+                    parent_data=sentence_text_object,
+                    instance_id=output_dataset_instance,
+                    input_language=input_language,
+                    output_language=output_language,
+                    input_text=input_sentence,
+                    machine_translation=translated_sentence,
+                    context=context,
+                    metadata_json=metadata,
+                )
+                translation_pair_obj.save()
```

Comment:
Bulk create would be more cheaper and efficient I suppose. Can you append these to an array, and after all sentences in a batch is complete, can you do `bulk_create`

Reply:
Turns out `bulk_create` only works for a non-inherited model. I get the following error when I use it to create a TranslationPair model 

```
ValueError at /functions/schedule_google_translate_job
Can't bulk create a multi-table inherited model
```

I found some fixes on StackOverflow but they are mostly [user developed hacks](https://stackoverflow.com/questions/49826482/django-valueerror-cant-bulk-create-a-multi-table-inherited-model) and currently only work for single inheritance level, apparently. It might also be risky to implement them because they could lead to errors in the future.

Reply:
> I found some fixes on StackOverflow

Thanks for sharing. Can we use this one?
https://stackoverflow.com/a/72939340/

Basically, shall we include this as a util, import it and use it wherever we want to `bulk_create` inherited models?
Do you see any drawbacks? (other than those mentioned below.

>  currently only work for single inheritance level
Yeah, even our models are mostly single inheritance level only.

> they could lead to errors in the future.
Yes, makes sense. But usually Django develops things in a way such that things are always backward-compatible.
So I feel it's fine. If required, if we face issues in next major version of Django, we can handle it.
What do you think?

Reply:
Please also check if auto-increment works (the prev answer to that answer has mentioned it does not work)

Reply:
I have tried this code. It's working and I have added the bulk create functionality. If we continue with single inheritance then I don't see any immediate issues. The autoincrement is also working since the TranslationPairs have been allotted appropriate IDs.

Reply:
If this fine then I will ask @ishvindersethi22 to test and merge the code.

Reply:
Thanks, looks good to me!